SCClassLibrary/Common/Core/Kernel.sc

```diff
@@ -452,6 +462,10 @@ FunctionDef {
 		_FunctionDefDumpContexts
 		^this.primitiveFailed
 	}
+
+	defaultArgs { ^this.prototypeFrame.keep(argNames.size).copy }
```

Comment:
Since this is a basic method, let's avoid double copying, `keep` returns a new array also if n = a.size:

```

a = [1, 2, 3];
b = a.keep(3);
a === b // false
```